pkg/engine/update.go

```diff
@@ -237,7 +237,7 @@ func installPlugins(
 		return nil, nil, err
 	}
 
-	allPlugins := languagePlugins.Union(snapshotPlugins)
+	allPlugins := languagePlugins.Union(snapshotPlugins).Deduplicate()
```

Comment:
computeDefaultProviderPlugins sorts and selects the default provider by greatest version anyway, why do we need to deduplicate?

Reply:
We need to deduplicate to avoid installing plugins which will never be used. I could move the `Deduplicate` call to only effect the set passed into `ensurePluginsAreInstalled` to limit the impact of the change. You are correct that it won't matter elsewhere.